config/externalname.go

```diff
@@ -2659,14 +2659,38 @@ var CLIReconciledExternalNameConfigs = map[string]config.ExternalName{
 	"aws_vpc_security_group_egress_rule": vpcSecurityGroupRule(),
 	// Imported by using the id: sgr-02108b27edd666983
 	"aws_vpc_security_group_ingress_rule": vpcSecurityGroupRule(),
-	// us-west-2_abc123/3ho4ek12345678909nh3fmhpko
-	"aws_cognito_user_pool_client": FormattedIdentifierFromProvider("", "name"),
+	// Cognito User Pool clients can be imported using the user pool id and client id separated by a slash (/)
+	// However, the terraform id is just the client id.
+	"aws_cognito_user_pool_client": cognitoUserPoolClient(),
 	// simpledb
 	//
 	// SimpleDB Domains can be imported using the name
 	"aws_simpledb_domain": config.NameAsIdentifier,
 }
 
+// cognitoUserPoolClient
+// Note(mbbush) This resource has some unexpected behaviors that make it impossible to write a completely correct
+// ExternalName config. Specifically, the terraform id returned in the terraform state is not the same as the
+// identifier used to import it. Additionally, if the terraform id set to an empty string, the terraform
+// provider passes the empty string through to the aws query during refresh, which returns an api error.
+// This could be related to the fact that this resource is implemented using the terraform plugin framework,
+// which introduces the concept of a null value as distinct from a zero value.
+func cognitoUserPoolClient() config.ExternalName {
+	e := config.IdentifierFromProvider
+	// TODO: Uncomment when it's acceptable to remove fields from spec.initProvider (major release)
+	//e.IdentifierFields = []string{"user_pool_id"}
```

Comment:
Before I rebased and included https://github.com/upbound/provider-aws/pull/1046, setting this was a non-breaking change, since reference fields and identifier fields were both omitted from `spec.initProvider`. Now, if I set it here, then upjet removes `userPoolId`, `userPoolIdRef` and `userPoolIdSelector` from `spec.initProvider`. 

For this resource, not having those fields in `initProvider` seems totally fine (if they change, terraform will require destroying and recreating the resource anyways), and it is necessary to know the user pool id in some form in order to read the state of the resource from AWS, so it seems to conceptually fit as an identifier field. 

This is a bugfix, so I think the best compromise is to leave it commented out, with a TODO to add it back when we do a major release.